Urgent: CMake strategy for dependencies fully broken for newer CMake

I have a `CMakeLists.txt` that includes
```cmake
set(HIGHFIVE_USE_XTENSOR 1)
set(HIGHFIVE_USE_BOOST 0)
find_package(HighFive REQUIRED)
```

Now I'm getting:
```
CMake Warning (dev) at /Users/tdegeus/opt/miniconda3/envs/test/share/HighFive/CMake/HighFiveConfig.cmake:50 (option):
  Policy CMP0077 is not set: option() honors normal variables.  Run "cmake
  --help-policy CMP0077" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  For compatibility with older versions of CMake, option is clearing the
  normal variable 'HIGHFIVE_USE_BOOST'.
Call Stack (most recent call first):
  CMakeLists.txt:7 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at /Users/tdegeus/opt/miniconda3/envs/test/share/HighFive/CMake/HighFiveConfig.cmake:52 (option):
  Policy CMP0077 is not set: option() honors normal variables.  Run "cmake
  --help-policy CMP0077" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

  For compatibility with older versions of CMake, option is clearing the
  normal variable 'HIGHFIVE_USE_XTENSOR'.
Call Stack (most recent call first):
  CMakeLists.txt:7 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

HIGHFIVE 2.2.2: (Re)Detecting Highfive dependencies (HIGHFIVE_USE_INSTALL_DEPS=NO)
-- HDF5 C compiler wrapper is unable to compile a minimal HDF5 program.
-- Found HDF5: /Users/tdegeus/opt/miniconda3/envs/test/lib/libhdf5.dylib (found version "1.10.6")
CMake Error at /Users/tdegeus/opt/miniconda3/envs/test/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:218 (message):
  Could NOT find Boost (missing: Boost_INCLUDE_DIR system serialization)
Call Stack (most recent call first):
  /Users/tdegeus/opt/miniconda3/envs/test/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:582 (_FPHSA_FAILURE_MESSAGE)
  /Users/tdegeus/opt/miniconda3/envs/test/share/cmake-3.19/Modules/FindBoost.cmake:2208 (find_package_handle_standard_args)
  /Users/tdegeus/opt/miniconda3/envs/test/share/HighFive/CMake/HighFiveTargetDeps.cmake:25 (find_package)
  /Users/tdegeus/opt/miniconda3/envs/test/share/HighFive/CMake/HighFiveConfig.cmake:67 (include)
  CMakeLists.txt:7 (find_package)
```

So there is no way to customise the dependencies anymore. I have to compile with Boost even if I entirely not use it.
See https://cmake.org/cmake/help/latest/policy/CMP0077.html

I know I have been complaining about this several times before, but now the time has truly come to move towards a modular HighFive target!

--- a/CMake/HighFiveConfig.cmake.in
+++ b/CMake/HighFiveConfig.cmake.in
@@ -35,35 +35,32 @@ endif()
 
 # If the user sets this flag, all dependencies are preserved.
 # Useful in central deployments where dependencies are not prepared later
-option(HIGHFIVE_USE_INSTALL_DEPS "Use original Highfive dependencies" @HIGHFIVE_USE_INSTALL_DEPS@)
+set(HIGHFIVE_USE_INSTALL_DEPS @HIGHFIVE_USE_INSTALL_DEPS@ CACHE BOOL "Use original Highfive dependencies")
 if(HIGHFIVE_USE_INSTALL_DEPS)
   # If enabled in the deploy config, request c++14
   if(@HIGHFIVE_USE_XTENSOR@ AND NOT CMAKE_VERSION VERSION_LESS 3.8)
     set_property(TARGET HighFive APPEND PROPERTY INTERFACE_COMPILE_FEATURES cxx_std_14)
   endif()
-  message("HIGHFIVE @PROJECT_VERSION@: Using original dependencies (HIGHFIVE_USE_INSTALL_DEPS=YES)")
+  message(STATUS "HIGHFIVE @PROJECT_VERSION@: Using original dependencies (HIGHFIVE_USE_INSTALL_DEPS=YES)")
   copy_interface_properties(HighFive HighFive_HighFive)
   return()
 endif()
 
 # When not using the pre-built dependencies, give user options
-option(HIGHFIVE_USE_BOOST "Enable Boost Support" ON)
-option(HIGHFIVE_USE_EIGEN "Enable Eigen testing" OFF)
-option(HIGHFIVE_USE_XTENSOR "Enable xtensor testing" OFF)
-
-# Options when used from external projects. Keep defaults
-if(NOT DEFINED HIGHFIVE_PARALLEL_HDF5)
-  option(HIGHFIVE_PARALLEL_HDF5 "Enable Parallel HDF5 support" @HIGHFIVE_PARALLEL_HDF5@)
+if(DEFINED HIGHFIVE_USE_BOOST)
+  set(HIGHFIVE_USE_BOOST ${HIGHFIVE_USE_BOOST} CACHE BOOL "Enable Boost Support")
+else()
+  set(HIGHFIVE_USE_BOOST @HIGHFIVE_USE_BOOST@ CACHE BOOL "Enable Boost Support")
 endif()
-if(NOT DEFINED HIGHFIVE_USE_BOOST)
-  option(HIGHFIVE_USE_BOOST "Enable Boost Support" @HIGHFIVE_USE_BOOST@)
-endif()
+set(HIGHFIVE_USE_EIGEN "${HIGHFIVE_USE_EIGEN}" CACHE BOOL "Enable Eigen testing")
+set(HIGHFIVE_USE_XTENSOR "${HIGHFIVE_USE_XTENSOR}" CACHE BOOL "Enable xtensor testing")
+set(HIGHFIVE_PARALLEL_HDF5 @HIGHFIVE_PARALLEL_HDF5@ CACHE BOOL "Enable Parallel HDF5 support")
 
 if(HIGHFIVE_USE_XTENSOR AND NOT CMAKE_VERSION VERSION_LESS 3.8)
   set_property(TARGET HighFive APPEND PROPERTY INTERFACE_COMPILE_FEATURES cxx_std_14)
 endif()
 
-message("HIGHFIVE @PROJECT_VERSION@: (Re)Detecting Highfive dependencies (HIGHFIVE_USE_INSTALL_DEPS=NO)")
+message(STATUS "HIGHFIVE @PROJECT_VERSION@: (Re)Detecting Highfive dependencies (HIGHFIVE_USE_INSTALL_DEPS=NO)")
 include("${CMAKE_CURRENT_LIST_DIR}/HighFiveTargetDeps.cmake")
 foreach(dependency HighFive_libheaders libdeps)
     copy_interface_properties(HighFive ${dependency})

